widgetssdk/src/main/java/com/glia/widgets/survey/SingleChoiceAdapter.java

```diff
@@ -0,0 +1,93 @@
+package com.glia.widgets.survey;
+
+import android.annotation.SuppressLint;
+import android.graphics.Color;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import androidx.annotation.LayoutRes;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.recyclerview.widget.RecyclerView;
+import com.glia.androidsdk.engagement.Survey;
+import com.glia.widgets.R;
+
+import java.util.List;
+
+public class SingleChoiceAdapter extends RecyclerView.Adapter<SingleChoiceAdapter.ViewHolder> {
+
+    private final QuestionItem questionItem;
+    private final SingleChoiceAdapter.SingleChoiceCallback scaleResultCallback;
+
+    public SingleChoiceAdapter(QuestionItem questionItem, SingleChoiceAdapter.SingleChoiceCallback scaleResultCallback) {
+        this.questionItem = questionItem;
+        this.scaleResultCallback = scaleResultCallback;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = createNewLayout(parent, R.layout.survey_single_choice_option);
+        return new SingleChoiceAdapter.ViewHolder(view);
+    }
+
+    private static View createNewLayout(ViewGroup parent, @LayoutRes int layoutId) {
+        return LayoutInflater.from(parent.getContext()).inflate(layoutId, parent, false);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        List<Survey.Question.Option> optionList = getOptions();
+        if (optionList == null) {
+            return;
+        }
+        Survey.Question.Option option = optionList.get(position);
+
+        holder.tvChoice.setText(option.getLabel());
+
+        Survey.Answer answer = questionItem.getAnswer();
+        if (answer != null && option.getId().equals(answer.getResponse())) {
+            holder.layoutChoice.setBackgroundResource(R.drawable.border_background);
+        } else {
+            holder.layoutChoice.setBackgroundColor(Color.TRANSPARENT);
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        List<Survey.Question.Option> optionList = getOptions();
+        return optionList == null ? 0 : optionList.size();
```

Comment:
would be nice to get immutable empty list instead of optionList = null but that's just a note :)

Reply:
It shouldn't be null. It just for safety.